lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -2138,6 +2138,89 @@ class DecomposeAtenNumpyTOp : public OpRewritePattern<AtenNumpyTOp> {
 };
 } // namespace
 
+template <typename OpTy>
+static LogicalResult calculateVariance(OpTy op, PatternRewriter &rewriter,
+                                       bool unbiased, int64_t correction) {
+  Location loc = op.getLoc();
+  Value self = op.self();
+  Value dimList = op.dim();
+  Value keepDim = op.keepdim();
+  Type outputType = op.getType();
+  BaseTensorType inputTensorTy = self.getType().cast<BaseTensorType>();
+  if (!inputTensorTy.hasDtype() ||
+      !inputTensorTy.getDtype().isa<mlir::FloatType>()) {
+    return rewriter.notifyMatchFailure(op, "support floating type input only");
+  }
+
+  unsigned inputRank = getTensorRank(self);
+  SmallVector<Value> dimListElements;
+  bool isNoneOrEmpty = true;
+  if (!dimList.getType().template isa<Torch::NoneType>()) {
+    if (!getListConstructElements(dimList, dimListElements))
+      return rewriter.notifyMatchFailure(
+          op, "expect dimList to be constructed from list construct");
+    if (dimListElements.size() != 0 || inputRank == 0)
+      isNoneOrEmpty = false;
```

Comment:
A couple of comments/questions on this:

1. Why is the `inputRank == 0` condition needed? If `inputRank == 0`, the `for` loop in line 2166 should not run, so the behavior should be the same.
2. You don't need the `if`. You can do `isNoneOrEmpty = dimListElements.empty();`

Reply:
For your first comment/question if we don't check for the condition `inputRank == 0`, then it's true that the for loop will not run but https://github.com/llvm/torch-mlir/blob/12ee0d041b15d501ca07d758fe647d62cd8cbc7e/lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp#L2169-L2171 will again create an empty list, which we already have.